docs/spec_draft.md

```diff
@@ -485,6 +486,67 @@ IEEE-754 specification.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.compare
+
+### Semantics
+
+Performs element-wise comparison of `lhs` and `rhs` tensors based on
+`comparison_direction` and produces a `result` tensor. More formally,
+`result[i0, ..., iR-1]` =
+`compare_function(lhs[i0, ..., iR-1], rhs[i0, ..., iR-1])`.
+
+### Inputs
+
+| Name                   | Type                          |
+|------------------------|-------------------------------|
+| `lhs`                  | tensor of any supported type  |
+| `rhs`                  | tensor of any supported type  |
+| `comparison_direction` | type of comparison to perform |
+
+### Outputs
+
+| Name     | Type                |
+|----------|---------------------|
+| `result` | tensor of type `i1` |
+
+### Constraints
+
+  * (C1) `lhs` and `rhs` have the same type.
```

Comment:
Here is a proposal for separating the constraints more clearly (element type of result is already in Outputs):
```
* (C1) `lhs` and `rhs` have the same element type.
* (C2) `lhs`, `rhs`, and `result` have the same shape.
```